### PR TITLE
[JENKINS-31842] Define status of a StepExecution

### PR DESCRIPTION
**Pull request by jglick**
Used by https://github.com/jenkinsci/workflow-cps-plugin/pull/25.

@reviewbybees

### CODE REVIEW

**Comment by svanoort** [comment]
🐝

**Comment by abayer** [comment]
:bee:

**Comment by kohsuke** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/workflow/steps/AbstractSynchronousNonBlockingStepExecution.java [hunk 1]
```diff
@@ -74,7 +75,7 @@ public void onResume() {
         getContext().onFailure(new Exception("Resume after a restart not supported for non-blocking synchronous steps"));
     }
 
-    @Override public String toString() {
+    @Override public @Nonnull String getStatus() {
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/AbstractSynchronousNonBlockingStepExecution.java]: So... we're overriding a method with CheckForNull with one that is Nonnull?  I take it that findbugs is smart enough to handle this case correctly?

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/AbstractSynchronousNonBlockingStepExecution.java]: It seems to be, yes.

**Pull request merged**